src/effect_on_condition.cpp

```diff
@@ -357,6 +357,23 @@ bool effect_on_condition::activate( dialogue &d, bool require_callstack_check )
     return retval;
 }
 
+bool effect_on_condition::activate_activation_only( dialogue &d, const std::string &text1,
+        const std::string &text2, const std::string &text3, bool require_callstack_check ) const
+{
+    if( type == eoc_type::ACTIVATION ) {
+        return activate( d, require_callstack_check );
+    }
+    debugmsg(
+        "Must use an activation eoc for " + text1 +
+        ( text2.empty() ?
+          "" : ( ".  If you don't want the effect_on_condition to happen on its own (without the " + text2 +
+                 "), remove the recurrence min and max." ) ) +
+        ".  Otherwise, create a non-recurring effect_on_condition for this " + text3 +
+        ( text3.empty() ? "" : " " ) +
+        "with its condition and effects, then have a recurring one queue it." );
```

Comment:
Use printf-style formatting, it's much more readable, especially when you're stuffing some nested formatting in.
```suggestion
    debugmsg( "Must use an activation eoc for %s%s.  Otherwise, create a non-recurring effect_on_condition for this %s with its condition and effects, then have a recurring one queue it.", 
    text1,
    text2.empty() ? "" : 
    string_format(".  If you don't want the effect_on_condition to happen on its own (without the %s), remove the recurrence min and max", text2 ),
    text3 );
```

Reply:
Did a subtly different version of this. Thanks for the suggestion.